### ISSUE TITLE
3dDNA

### ISSUE THREAD

**Issue by jonnyPizza** [ubccr/software-layer#347]
This software is used by many bioinformatic groups here at UB. Used mostly for genome assembly. Please install the following softwares:

https://github.com/aidenlab/3d-dna
(I believe there is an EasyBuild for this one.)

Thank you very much

**Comment by dsajdak** [comment]
@jonnyPizza would you please separate these into 2 separate issues as we are trying to share the load amongst staff?  Thanks.  FYI the EB recipe for 3d-dna is using a very old toolchain but we'll try to get it working with a toolchain CCR supports.  Anything without an EB recipe falls to the bottom of the to-do list but we'll leave it on the backlog list for when time permits.

**Comment by tonykew** [comment]
3d-dna version 201008has been built, and is going through internal validation,
however we have no way to check if the software works as expected.
I will update this request once 3d-dna has been published.

**Comment by tonykew** [comment]
3d-dna version 201008 has been published to the ccrsoft/2023.01 software release.

```
login1$ module spider 3d-dna

----------------------------------------------------------------------------
  3d-dna:
----------------------------------------------------------------------------
    Description:
      3D de novo assembly (3D DNA) pipeline

     Versions:
        3d-dna/201008-Python-3.9.6
[...]
login1$ module spider 3d-dna/201008-Python-3.9.6

----------------------------------------------------------------------------
  3d-dna: 3d-dna/201008-Python-3.9.6
----------------------------------------------------------------------------
    Description:
      3D de novo assembly (3D DNA) pipeline


    You will need to load all module(s) on any one of the lines below before the "3d-dna/201008-Python-3.9.6" module is available to load.

      gcccore/11.2.0
[...]
login1$ module load gcccore/11.2.0 3d-dna/201008-Python-3.9.6 
login1$ 
```

This module includes two scripts: "run-asm-pipeline.sh" and
"run-asm-pipeline-post-review.sh"

```
login1$ run-asm-pipeline.sh
/cvmfs/soft.ccr.buffalo.edu/versions/2023.01/easybuild/software/avx512/Compiler/gcccore/11.2.0/3d-dna/201008-Python-3.9.6/run-asm-pipeline.sh 
version: 190716
Not sure how to parse your input: files not listed or not found at expected locations. Exiting!

*****************************************************
3D de novo assembly: version 190716

USAGE: ./run-asm-pipeline.sh [options] <path_to_input_fasta> <path_to_input_mnd> 

DESCRIPTION:
This is a script to assemble draft assemblies (represented in input by draft fasta and deduplicated list of alignments of Hi-C reads to this fasta as produced by the Juicer pipeline) into chromosome-length scaffolds. The script will produce an output fasta file, a Hi-C map of the final assembly, and a few supplementary annotation files to help review the result in Juicebox.

ARGUMENTS:
path_to_input_fasta			Specify file path to draft assembly fasta file.
path_to_input_mnd			Specify path to deduplicated list of alignments of Hi-C reads to the draft assembly fasta as produced by the Juicer pipeline: the merged_nodups file (mnd).

OPTIONS:
-m|--mode haploid/diploid			Runs in specific mode, either haploid or diploid (default is haploid).
-i|--input input_size			Specifies threshold input contig/scaffold size (default is 15000). Contigs/scaffolds smaller than input_size are going to be ignored.
-r|--rounds number_of_edit_rounds			Specifies number of iterative rounds for misjoin correction (default is 2).
-s|--stage stage					Fast forward to later assembly steps, can be polish, split, seal, merge and finalize.
-h|--help			Shows this help. Type --help for a full set of options.
*****************************************************

login1$ run-asm-pipeline-post-review.sh 
/cvmfs/soft.ccr.buffalo.edu/versions/2023.01/easybuild/software/avx512/Compiler/gcccore/11.2.0/3d-dna/201008-Python-3.9.6/run-asm-pipeline-post-review.sh 
Not sure how to parse your input: files not listed or not found at expected locations. Exiting!

*****************************************************
3D de novo assembly: version 180114

USAGE: ./run-asm-pipeline-post-review.sh [options] -r <review.assembly> <path_to_input_fasta> <path_to_input_mnd> 

DESCRIPTION:
This is a script to finalize assemblies (represented in input by draft fasta and deduplicated list of alignments of Hi-C reads to this fasta as produced by the Juicer pipeline) into chromosome-length fasta sequences, after review in Juicebox Assembly Tools Module (represented by review.assembly file). The script will produce an output fasta file, a Hi-C map of the final assembly, and a few supplementary annotation files to help review the result in Juicebox.

ARGUMENTS:
path_to_input_fasta			Specify file path to draft assembly fasta file.
path_to_input_mnd			Specify path to deduplicated list of alignments of Hi-C reads to the draft assembly fasta as produced by the Juicer pipeline: the merged_nodups file (mnd).

OPTIONS:
-r|--review path_to_review_assembly
			Path to review ".assembly" file.
-i|--input input_size
			Specifies threshold input contig/scaffold size (default is 15000). Contigs/scaffolds smaller than input_size are going to be ignored. Only matters if running including seal. Should be the same as used for running the original script.
-s|--stage stage
			Assembly steps to run on top of the reviewed assembly, can be seal and finalize. Default is finalize.
-q|--mapq mapq
			Mapq threshold for final map(s) visualization, default is 1.
-g|--gap_size gap_size
			Gap size to be added between scaffolded sequences in the final chromosome-length scaffolds (default is 500).	
--sort-output
			Option to sort the chromosome-length scaffolds by size, in the descending order.
-c|--chromosome-map chromosome_count
			Option to output the standard Juicebox contact map with data sandboxed to chromosomes, corresponding to the _HiC.assembly after the gaps have been added between scaffolded sequences. Only the first chromosome_count sequences in the assembly will be included into the map.
-h|--help			Shows this help. Type --help for a full set of options.

*****************************************************

login1$ 

```
